app/views/contexts/preview/preview_question.py

```diff
@@ -0,0 +1,131 @@
+        )
+        self.answer_guidance = self._build_answer_guidance(
+            answers=iter(question_schema["answers"])
+        )
+        self.survey_data = survey_data
+
+    def _build_answers(
+        self, question_schema: QuestionSchemaType, survey_data: ImmutableDict[str, str]
+    ) -> list[Optional[str]]:
+        answers = []
+        for answer in iter(question_schema["answers"]):
+            if options := answer.get("options"):
+                for option in options:
+                    if isinstance(option["label"], dict):
+                        label = self.resolve_text(option["label"], survey_data)
+                        answers.append(label)
+                    else:
+                        answers.append(option["label"])
+            if not options:
+                answers.append(answer["label"])
+
+        return answers
+
+    @staticmethod
+    def _build_answer_descriptions(answers: Iterator[dict]) -> Optional[dict]:
+        return next(
+            (
+                answer.get("description")
+                for answer in answers
+                if answer.get("description")
+            ),
+            None,
+        )
+
+    def _build_answer_guidance(self, answers: Iterator[dict]) -> Optional[list[Any]]:
+        for answer in answers:
+            return self._build_guidance(answer)
+
+    def _build_descriptions(
+        self, question_schema: QuestionSchemaType, survey_data: ImmutableDict[str, str]
+    ) -> Any:  # QuestionnaireSchema.get_mutable_deepcopy returns "Any"
+        if descriptions := question_schema.get("description"):
+            mutable_descriptions = QuestionnaireSchema.get_mutable_deepcopy(
+                descriptions
+            )
+            for index, _ in enumerate(mutable_descriptions):
+                if isinstance(mutable_descriptions[index], dict):
+                    mutable_descriptions[index] = self.resolve_text(
+                        mutable_descriptions[index], survey_data
+                    )
+
+            return mutable_descriptions
+
+        return None
+
+    def _build_question_guidance(
+        self, question_schema: QuestionSchemaType
+    ) -> Optional[list[Any]]:
+        return self._build_guidance(question_schema)
+
+    @staticmethod
+    def _build_guidance(schema_element: QuestionSchemaType) -> Optional[list[dict]]:
+        if guidance := schema_element.get("guidance"):
+            guidance_list = []
+            for contents in guidance.get("contents"):
+                if contents.get("description"):
+                    guidance_list.append(contents.get("description"))
+                elif contents.get("list"):
+                    guidance_items = list(contents.get("list"))
+                    guidance_list.append(guidance_items)
+            return guidance_list
+        return None
+
+    @staticmethod
+    def _get_length(
+        answers: dict,
+    ) -> Optional[Any]:
+        return next(
```

Comment:
What if you have more than one?

Reply:
At the moment, if you add another text area they both show up as bullet points list and no info about length is added.